<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>gPAM50 Tutorial • gPAM50</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="gPAM50 Tutorial">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gPAM50</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gPAM50_tutorial.html">gPAM50 Tutorial</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://pages.github.roche.com/changc25/gPAM50">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><script src="gPAM50_tutorial_files/htmlwidgets-1.2/htmlwidgets.js"></script><script src="gPAM50_tutorial_files/jquery-1.12.4/jquery.min.js"></script><link href="gPAM50_tutorial_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="gPAM50_tutorial_files/datatables-binding-0.4/datatables.js"></script><link href="gPAM50_tutorial_files/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="gPAM50_tutorial_files/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="gPAM50_tutorial_files/dt-core-1.10.16/js/jquery.dataTables.min.js"></script><link href="gPAM50_tutorial_files/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet">
<script src="gPAM50_tutorial_files/crosstalk-1.0.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>gPAM50 Tutorial</h1>
                        <h4 class="author">Andrew Chang, Ching-Wei Chang, &amp; Paul Paczuski</h4>
            
            <h4 class="date">2018-10-17</h4>
      
      
      <div class="hidden name"><code>gPAM50_tutorial.Rmd</code></div>

    </div>

    
    
<div id="install-gpam50" class="section level2">
<h2 class="hasAnchor">
<a href="#install-gpam50" class="anchor"></a>0. Install gPAM50</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># install.packages("devtools")</span>
devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"changc25/gPAM50"</span>,
                          <span class="dt">host =</span> <span class="st">"https://github.roche.com/api/v3"</span>)</code></pre></div>
</div>
<div id="load-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>1. Load required packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(tidyverse)
<span class="kw">library</span>(gPAM50)
<span class="kw">library</span>(Biobase) ## for parsing ExpressionSet data
<span class="kw">library</span>(DT) ## this is optional. It helps render dataframe nicely in html
<span class="co"># library(edgeR) ## for normalizing RNAseq data (method: cpm). This is optional</span></code></pre></div>
</div>
<div id="load-example-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-example-data" class="anchor"></a>2. Load example data</h2>
<p>Subset of TCGA-BRCA. The data (object name is <code>es</code>) is pre-loaded when gPAM50 is called.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">es1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/Biobase/topics/assayData">assayData</a></span>(es)<span class="op">$</span>exprs)
<span class="co">#es1 = data.frame(cpm(es1)) ## raw count =&gt; CPM</span>
es1 &lt;-<span class="st"> </span><span class="kw">log2</span>(es1<span class="op">+</span><span class="dv">1</span>) ## log2 transform 
ph1 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/Biobase/topics/phenoData">pData</a></span>(es)
an1 &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/Biobase/topics/featureData">fData</a></span>(es)

<span class="kw">cat</span>(<span class="kw">dim</span>(es1)[<span class="dv">1</span>], <span class="st">"genes</span><span class="ch">\n</span><span class="st">"</span>)
<span class="co">#&gt; 20133 genes</span>
<span class="kw">cat</span>(<span class="kw">dim</span>(es1)[<span class="dv">2</span>], <span class="st">"samples</span><span class="ch">\n</span><span class="st">"</span>)
<span class="co">#&gt; 50 samples</span>
<span class="kw">cat</span>(<span class="kw">dim</span>(ph1)[<span class="dv">2</span>], <span class="st">"features</span><span class="ch">\n</span><span class="st">"</span>)
<span class="co">#&gt; 131 features</span></code></pre></div>
<div id="expression-data" class="section level3">
<h3 class="hasAnchor">
<a href="#expression-data" class="anchor"></a>2.1 Expression data</h3>
<p>Here are the first few records of the gene expression data. This is after a <code>log2</code> transformation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(es1[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/formatCurrency">formatRound</a></span>(<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>), <span class="dt">digits =</span> <span class="dv">3</span>)</code></pre></div>
<div id="htmlwidget-87e626fb88bb25b260a0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-87e626fb88bb25b260a0">{"x":{"filter":"none","data":[["GeneID:1","GeneID:10","GeneID:100","GeneID:1000","GeneID:10000"],[9.29920801838728,6.75488750216347,8.78135971352466,7.94251450533924,9.53138146051631],[10.7607199474656,2.32192809488736,9.29462074889163,5.58496250072116,7.8703647195834],[9.01680828768655,3.8073549220576,9.41574176829009,7.76155123244448,8.98013957763916],[9.39231742277876,2.8073549220576,9.59991284218713,8.26209484537018,9.02790599656988],[9.55266909751427,6,8.07681559705083,4.16992500144231,9.67595703294175],[8.62205181945638,3,9.13442632022093,12.7494504655699,9.17492568250068],[6.85798099512757,1.58496250072116,5.70043971814109,6.7279204545632,6.95419631038687],[8.86727873970966,4.85798099512757,8.30833903013941,9.09011241966429,9.32418054661874],[9.29920801838728,4.95419631038687,8.96000193206808,7.10852445677817,10.9255544398736],[9.68825030913318,5.16992500144231,7.54689445988764,6.16992500144231,7.98868468677217]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>TCGA.A2.A3KC.01A.11R.A213.07<\/th>\n      <th>TCGA.AR.A2LJ.01A.12R.A19W.07<\/th>\n      <th>TCGA.BH.A2L8.01A.11R.A18M.07<\/th>\n      <th>TCGA.C8.A3M8.01A.11R.A213.07<\/th>\n      <th>TCGA.E9.A3HO.01A.11R.A213.07<\/th>\n      <th>TCGA.HN.A2NL.01A.11R.A18M.07<\/th>\n      <th>TCGA.A1.A0SB.01A.11R.A144.07<\/th>\n      <th>TCGA.A1.A0SD.01A.11R.A115.07<\/th>\n      <th>TCGA.A1.A0SE.01A.11R.A084.07<\/th>\n      <th>TCGA.A1.A0SF.01A.11R.A144.07<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 1, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 2, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 7, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 8, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 9, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 10, 3, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="annotation-data" class="section level3">
<h3 class="hasAnchor">
<a href="#annotation-data" class="anchor"></a>2.2 Annotation data</h3>
<p>Gene information. Especially the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3013746/">EntrezID</a> is important for telling <code>gPAM50</code> package which gene is which. It is a unique ID for each gene. Some genes may have the same Symbol, but they will have different EntrezID. Make sure your dataset contains this information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(an1[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ])</code></pre></div>
<div id="htmlwidget-c888180cdb3d0ad0b554" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c888180cdb3d0ad0b554">{"x":{"filter":"none","data":[["GeneID:1","GeneID:10","GeneID:100","GeneID:1000","GeneID:10000"],["1","10","100","1000","10000"],["A1BG","NAT2","ADA","CDH2","AKT3"],["19","8","20","18","1"],[58346806,18391245,44619519,27950966,243488233],[58353499,18401213,44651735,28177481,243851079],[-1,1,-1,-1,-1],["alpha-1-B glycoprotein","N-acetyltransferase 2 (arylamine N-acetyltransferase)","adenosine deaminase","cadherin 2, type 1, N-cadherin (neuronal)","v-akt murine thymoma viral oncogene homolog 3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>EntrezID<\/th>\n      <th>Symbol<\/th>\n      <th>chr<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>str<\/th>\n      <th>GeneName<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="phenotype-data" class="section level3">
<h3 class="hasAnchor">
<a href="#phenotype-data" class="anchor"></a>2.3 Phenotype data</h3>
<p>This dataframe is not necessary for making prediction. Phenotype data describes the patient metadata, including some basic clinical features like gender and cancer status.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(ph1[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</code></pre></div>
<div id="htmlwidget-f9cf2adb6eb76de431a8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f9cf2adb6eb76de431a8">{"x":{"filter":"none","data":[["TCGA-A2-A3KC-01A-11R-A213-07","TCGA-AR-A2LJ-01A-12R-A19W-07","TCGA-BH-A2L8-01A-11R-A18M-07"],[null,null,null],["1","1","1"],["TCGA-A2-A3KC-01A-11R-A213-07","TCGA-AR-A2LJ-01A-12R-A19W-07","TCGA-BH-A2L8-01A-11R-A18M-07"],["TCGA-A2-A3KC","TCGA-AR-A2LJ","TCGA-BH-A2L8"],["tumor","tumor","tumor"],["15677","15980","15733"],["YES","NO","YES"],["NO","YES","NO"],[-20401,-14723,-16658],["FEMALE","FEMALE","FEMALE"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sizeFactor<\/th>\n      <th>condition<\/th>\n      <th>Sample_ID<\/th>\n      <th>Patient_ID<\/th>\n      <th>Cancer_Status<\/th>\n      <th>form_completion_date<\/th>\n      <th>prospective_collection<\/th>\n      <th>retrospective_collection<\/th>\n      <th>birth_days_to<\/th>\n      <th>gender<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="pam50-prediction" class="section level2">
<h2 class="hasAnchor">
<a href="#pam50-prediction" class="anchor"></a>3. PAM50 prediction</h2>
<p><code>gPAM50</code> takes two arguments:</p>
<ol style="list-style-type: decimal">
<li><p>expression dataframe; this is the dataframe shown above (section 2.1). Columns are different samples and rows are different genes. <strong>It is important to re-assign rownames of this dataframe to corresponding EntrezID.</strong> You can assign unique gene symbols to rownames as well, however, the symbol name may change depending on the RNA-seq platform, but the Entrez ID should be consistent.</p></li>
<li><p><code>cutoff</code>; the default predicted subtype is simply the one with maximum probability, you can also specify a custom cutoff with the <code>cutoff</code> parameter. For example, if <code>cutoff = 0.9</code>, only the subtype with &gt; 0.9 probability and is the max will be assigned to final PAM50 subtype call. If none of the subtypes produce probability &gt; 0.9, the final PAM50 call will be <code>NA</code>.</p></li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## IMPORTANT! Assign Gene ID to the expression data (es1)
<span class="kw">rownames</span>(es1) &lt;-<span class="st"> </span>an1<span class="op">$</span>EntrezID

## get prediction
df.pred &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gPAM50.html">gPAM50</a></span>(es1, <span class="dt">cutoff =</span> <span class="dv">0</span>)

DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">datatable</a></span>(<span class="kw">head</span>(df.pred)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>DT<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/DT/topics/formatCurrency">formatRound</a></span>(<span class="kw">c</span>(<span class="dv">2</span><span class="op">:</span><span class="dv">6</span>), <span class="dt">digits =</span> <span class="dv">3</span>)</code></pre></div>
<div id="htmlwidget-02ddf408d5c587435e70" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-02ddf408d5c587435e70">{"x":{"filter":"none","data":[["1","2","3","4","5","6"],["LumA","LumA","LumA","LumB","LumB","Basal"],[2.93179289693771e-13,3.11243578494545e-09,2.25147479855096e-13,7.9693032348621e-13,9.50997721630793e-10,0.999999999990971],[2.2065206128302e-06,0.00272886559996556,2.86904610198217e-05,0.0010132732784747,0.00261591108507397,9.0135325815298e-12],[0.99159463527834,0.792693433881244,0.997398812929999,0.0497278260000336,0.0502811384730608,2.70368725109662e-19],[0.000316694760614062,0.00139444955315713,0.00164447782112743,0.949258767594249,0.947100559954138,1.50427533860749e-14],[0.0080864634401397,0.203183247853197,0.000928018787628398,1.33126446153964e-07,2.38953672939941e-06,5.98012624048927e-16]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PAM50<\/th>\n      <th>Basal<\/th>\n      <th>Her2<\/th>\n      <th>LumA<\/th>\n      <th>LumB<\/th>\n      <th>Normal<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 2, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 3, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 3, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>It returns a dataframe with prediction result; the row order is the same as your input data. The <code>PAM50</code> column shows the predicted subtype. The other columns, <code>Basal</code>, <code>Her2</code>, <code>LumA</code>, <code>LumB</code>, and <code>Normal</code>, indicate the probability of each subtype.</p>
</div>
<div id="visualize-prediction-result" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-prediction-result" class="anchor"></a>4. Visualize prediction result</h2>
<p>There two methods for us to quickly visualize whether our prediction makes sense:</p>
<ol style="list-style-type: decimal">
<li>
<code>plotPAM50</code> Look at the heatmap to see if predicted subgroups have similar gene patterns as <a href="https://www.ncbi.nlm.nih.gov/core/lw/2.0/html/tileshop_pmc/tileshop_pmc_inline.html?title=Click%20on%20image%20to%20zoom&amp;p=PMC3&amp;id=2667820_zlj9990983710005.jpg">Parker’s paper</a>. In the heatmap, the order of those 50 genes are the same as Parker’s paper.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotPAM50.html">plotPAM50</a></span>(es1, df.pred<span class="op">$</span>PAM50)</code></pre></div>
<p><img src="gPAM50_tutorial_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
<p>2.<code>plotValidationGenes</code> Look at 4 well-known genes (can add more genes you know that also have significant patterns for each subgroup):</p>
<ul>
<li>
<code>ERBB2</code>: Her2 gene; Her2 group should have the highest ERBB2 value.</li>
<li>
<code>ESR1</code> : ER gene; LumA and LumB groups should have the highest ESR1 value.</li>
<li>
<code>MKI67</code>: tumor growth factor; LumA should have the lowest MKI67 value because LumA patients usually have the best prognosis.</li>
<li>
<code>PGR</code> : PR gene; LumA should have the highest PGR value.</li>
</ul>
<p>You can also visualize any other gene of interest by using the optional argument <code>genes = c("EGFR", "FOXA1")</code>. Note: this funciton currently only supports plotting 50 PAM50 genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotValidationGenes.html">plotValidationGenes</a></span>(es1, df.pred<span class="op">$</span>PAM50)</code></pre></div>
<p><img src="gPAM50_tutorial_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#install-gpam50">0. Install gPAM50</a></li>
      <li><a href="#load-required-packages">1. Load required packages</a></li>
      <li><a href="#load-example-data">2. Load example data</a></li>
      <li><a href="#pam50-prediction">3. PAM50 prediction</a></li>
      <li><a href="#visualize-prediction-result">4. Visualize prediction result</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://github.roche.com/changc25">Andrew Chang</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
